File: backend/app/schemas/fundamental.py
"""基本面分析请求与响应 Schema。"""

from datetime import date as dt_date
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field


class FundamentalQueryParams(BaseModel):
    page: int = Field(default=1, ge=1, description="页码")
    page_size: int = Field(default=20, ge=1, le=100, description="每页条数")
    code: str | None = Field(default=None, description="股票代码模糊匹配")
    name: str | None = Field(default=None, description="股票名称模糊匹配")
    min_roe: float | None = Field(default=None, description="ROE 最小值")
    max_roe: float | None = Field(default=None, description="ROE 最大值")
    min_debt_to_assets: float | None = Field(default=None, description="资产负债率最小值")
    max_debt_to_assets: float | None = Field(default=None, description="资产负债率最大值")
    sort_by: str | None = Field(default=None, description="排序字段")
    sort_order: Literal["asc", "desc"] = Field(default="desc", description="排序方向")


class FundamentalItem(BaseModel):
    code: str
    name: str | None = None
    exchange: str | None = None
    market: str | None = None
    report_date: dt_date | None = None
    ann_date: dt_date | None = None
    revenue: Decimal | None = None
    net_profit: Decimal | None = None
    eps: Decimal | None = None
    bps: Decimal | None = None
    roe: Decimal | None = None
    roe_dt: Decimal | None = None
    roe_waa: Decimal | None = None
    roa: Decimal | None = None
    gross_margin: Decimal | None = None
    net_margin: Decimal | None = None
    debt_to_assets: Decimal | None = None
    current_ratio: Decimal | None = None
    quick_ratio: Decimal | None = None
    cfps: Decimal | None = None
    ebit: Decimal | None = None
    ocf_to_profit: Decimal | None = None

    model_config = {"from_attributes": True}


class FundamentalResponse(BaseModel):
    items: list[FundamentalItem] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 20

File: frontend/src/api/fundamental.ts
import http from './http'

export interface FundamentalItem {
  code: string
  name: string | null
  exchange: string | null
  market: string | null
  report_date: string | null
  ann_date: string | null
  revenue: number | null
  net_profit: number | null
  eps: number | null
  bps: number | null
  roe: number | null
  roe_dt: number | null
  roe_waa: number | null
  roa: number | null
  gross_margin: number | null
  net_margin: number | null
  debt_to_assets: number | null
  current_ratio: number | null
  quick_ratio: number | null
  cfps: number | null
  ebit: number | null
  ocf_to_profit: number | null
}

export interface FundamentalResponse {
  items: FundamentalItem[]
  total: number
  page: number
  page_size: number
}

export function getFundamentals(params: {
  page?: number
  page_size?: number
  code?: string
  name?: string
  min_roe?: number
  max_roe?: number
  min_debt_to_assets?: number
  max_debt_to_assets?: number
  sort_by?: string
  sort_order?: 'asc' | 'desc'
}) {
  return http.get<FundamentalResponse>('/fundamental/analysis', { params })
}
